### PR TITLE
Search for cards in the debug-ui using the description as a filter.

### PR DESCRIPTION
**Pull request by kberg**
To make this work I pre-fetch all the cards, and store data in memory.
This turns out to make searching MUCH faster!

Incidentally, most of the cards do not have text, so while this will seem broken, it surely isn't.

### CODE REVIEW

File: src/components/DebugUI.ts [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
 } from '../cards/AllCards';
 import {GameModule} from '../GameModule';
+import {ICard} from '../cards/ICard';
+
+interface Entry {
+  card: ICard,
+  module: GameModule
+}
+const cards: Map<string, Entry> = new Map();
+ALL_CARD_MANIFESTS.forEach((manifest) => {
```
Comment (chosta) [inline src/components/DebugUI.ts:17]: If I'm reading this correct, there are three identical `.forEach` loops doing the same thing? Is there any hidden purpose to it?

File: src/components/DebugUI.ts [hunk 2]
```diff
@@ -48,15 +65,22 @@ export const DebugUI = Vue.component('debug-ui', {
...
-      if (!nameMatches) {
-        return false;
+      const card = cards.get(cardName);
+      const filterText = this.$data.filterText.toUpperCase();
+      if (this.$data.filterText.length > 0) {
+        if (cardName.toUpperCase().indexOf(filterText) === -1) {
+          return false;
+        }
+        if (this.$data.filterDescription) {
+          const desc = card?.card.metadata?.description;
```
Comment (chosta) [inline src/components/DebugUI.ts:75]: I think we can also search through all text inside the card too. I'll write a function for CardRenderer for it.

**Review by chosta — APPROVED** [review]
Looks good, just not sure about the identical `forEach` loops and I'm adding a TODO for myself to have a CardRender.getAllText to be used for the global search.

File: src/components/DebugUI.ts (hunk 1 above)
Comment (kberg) [reply src/components/DebugUI.ts:17]: I tried joining them as one, but it was a typescript mess.

File: src/components/DebugUI.ts (hunk 2 above)
Comment (kberg) [reply src/components/DebugUI.ts:75]: Isn't that what this is?

Comment (kberg) [reply src/components/DebugUI.ts:75]: (What did I get wrong?)

**Pull request merged**